Modules/User/Services/UserUpdateService.php

```diff
@@ -29,9 +29,12 @@ public function __construct(UserRepositoryInterface $userRepo) {
      * @return User|null
      */
     public function update(array $data) : ?User {
-        $data["password"] = Hash::make($data["password"]);
-        $this->userRepo->update($data, $data["id"]);
-        $user = $this->userRepo->findBy("id", $data["id"]); //$ids es un arreglo de ids $ids = [1, 2, 3, 4]
+        if(isset($data["password"]) && $data["password"] != ""){
+            $data["password"] = Hash::make($data["password"]);
+        }else{
+            unset($data["password"]);
+        }        
+        $user = $this->userRepo->updateAndReturn($this->userRepo->findBy("id", $data["id"]), $data);
```

Comment:
$user = $this->userRepo->updateAndReturn($data['id'], $data);

Reply:
Ya lo intenté así como me dices pero me marca error, por eso lo puse así:
$user = $this->userRepo->updateAndReturn($this->userRepo->findBy("id", $data["id"]), $data);